### PR TITLE
Log path in a particular BIFE

### PR DESCRIPTION
**Pull request by danmosemsft**
before
```
System.BadImageFormatException : Bad IL format.
```
after
```
System.BadImageFormatException : Bad IL format. The format of the file 'C:\WINDOWS\system32\kernelbase.dll' is invalid.
```

Note
* Just some of countless error paths for assembly load. (Over 60 instances of THROW_BAD_FORMAT alone. Some have paths some do not.)
* I can only hit the 1st path in tests. I can revert the other 3 if you like.

### CODE REVIEW

File: src/vm/excep.cpp [hunk 1]
```diff
@@ -12327,20 +12327,25 @@ VOID ThrowBadFormatWorker(UINT resID, LPCWSTR imageName DEBUGARG(__in_z const ch
 #ifndef DACCESS_COMPILE
     SString msgStr;
 
-    if ((imageName != NULL) && (imageName[0] != 0))
```
Comment (jkotas) [inline src/vm/excep.cpp]: Could you please also try some of the cases that went through this path before your change - to make sure that they look as good or better?

Comment (danmosemsft) [reply src/vm/excep.cpp]: I did some experimentation in the debugger. It just moves the path from the start into the message. I have fixed a mistake in the fallback path, though.

**Comment by jkotas** [comment]
The managed formatting puts the filename on a new line: https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/BadImageFormatException.cs#L99 . Should the managed and unmanaged formatting be in sync?

**Comment by jkotas** [comment]
Ideally, we would also make sure that the BadImageFormatException.FileName property is set properly, but that may be too hard to do.

**Comment by danmosemsft** [comment]
> Should the managed and unmanaged formatting be in sync

It’s useful and important that the file name be in the message, as sometimes the message is all you have.  Where the exception type does not have a FileName property, it certainly the case it has to be in the message, and that is the case for eg IOException which by its nature will typically be associated with a file.

So to me the useful and consistent pattern is to have it in the message.  (In fact I had proposed a change earlier to FNFE to remove the file name from ToString if it was already in the message, but @stephentoub pointed out that someone might be parsing it.)

**Comment by danmosemsft** [comment]
> Ideally, we would also make sure that the BadImageFormatException.FileName property is set properly, but that may be too hard to do.

Can we defer this for now? I am not introducing that.

**Comment by jkotas** [comment]
ok

**Pull request merged**